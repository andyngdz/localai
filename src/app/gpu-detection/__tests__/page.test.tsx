import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import GPUDetectionScreen from "../page";

// Mock the GpuDetection component
vi.mock("@/features/gpu-detection", () => ({
  GpuDetection: () => (
    <div data-testid="gpu-detection">GPU Detection Component</div>
  ),
}));

describe("GPUDetectionScreen", () => {
  it("renders GpuDetection component", () => {
    render(<GPUDetectionScreen />);

    expect(screen.getByTestId("gpu-detection")).toBeInTheDocument();
    expect(screen.getByText("GPU Detection Component")).toBeInTheDocument();
  });

  it("returns GpuDetection as default export", () => {
    const { container } = render(<GPUDetectionScreen />);

    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByTestId("gpu-detection")).toBeInTheDocument();
  });

  it("has correct component structure", () => {
    const { container } = render(<GPUDetectionScreen />);

    // Should render only the GpuDetection component
    expect(container.children).toHaveLength(1);
    expect(screen.getByTestId("gpu-detection")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<GPUDetectionScreen />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
