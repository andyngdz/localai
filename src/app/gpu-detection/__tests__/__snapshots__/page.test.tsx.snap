// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GPUDetectionScreen > matches snapshot 1`] = `
<div
  data-testid="gpu-detection"
>
  GPU Detection Component
</div>
`;
